domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/Agency.java

```diff
@@ -0,0 +1,275 @@
+         * @return builder for future object creation
+         */
+        public AgencyBuilder agencyId(@Nullable final String agencyId) {
+            this.agencyId = agencyId;
+            return this;
+        }
+
+        /**
+         * Sets field agencyLang value and returns this
+         *
+         * @param agencyLang primary language used by this transit agency
+         * @return builder for future object creation
+         */
+        public AgencyBuilder agencyLang(@Nullable final String agencyLang) {
+            this.agencyLang = agencyLang;
+            return this;
+        }
+
+        /**
+         * Sets field agencyPhone value and returns this
+         *
+         * @param agencyPhone a voice telephone number for the specified agency
+         * @return builder for future object creation
+         */
+        public AgencyBuilder agencyPhone(@Nullable final String agencyPhone) {
+            this.agencyPhone = agencyPhone;
+            return this;
+        }
+
+        /**
+         * Sets field agencyFareUrl value and returns this
+         *
+         * @param agencyFareUrl URL of a web page that allows a rider to purchase tickets or other fare instruments for
+         *                      that agency online         * @return builder for future object creation
+         */
+        public AgencyBuilder agencyFareUrl(@Nullable final String agencyFareUrl) {
+            this.agencyFareUrl = agencyFareUrl;
+            return this;
+        }
+
+        /**
+         * Sets field agencyEmail value and returns this
+         *
+         * @param agencyEmail email address actively monitored by the agency’s customer service department
+         * @return builder for future object creation
+         */
+        public AgencyBuilder agencyEmail(@Nullable final String agencyEmail) {
+            this.agencyEmail = agencyEmail;
+            return this;
+        }
+
+        /**
+         * Returns list of notice generated when building an {@code Agency} from this.
+         *
+         * @return the list of notice generated when building an {@link Agency} from this.
+         */
+        public List<Notice> getNoticeCollection() {
+            return noticeCollection;
+        }
+
+        /**
+         * Returns a {@link Agency} object from fields provided via {@link AgencyBuilder} methods.
+         * Adds {@code MissingRequiredValueNotice} to notice collection  if fields agencyName, agencyUrl, or
+         * agencyTimezone are null. If said fields are null, method returns null
+         *
+         * @return Entity representing a row from agency.txt if the requirements from the official GTFS specification
+         * are met. Otherwise, method returns null.
+         */
+        public Agency build() {
+            noticeCollection.clear();
+            if (agencyName == null || agencyUrl == null || agencyTimezone == null) {
```

Comment:
return null after each if statement instead of a global if statement returning null value